R/wrap_vfold.R

```diff
@@ -0,0 +1,218 @@
+#' V-Fold Cross-Validation with Buffering
+#'
+#' V-fold cross-validation (also known as k-fold cross-validation) randomly
+#' splits the data into V groups of roughly equal size (called "folds").
+#' A resample of the analysis data consists of V-1 of the folds while the
+#' assessment set contains the final fold.
+#' This function adds the ability to apply an inclusion radius and exclusion
+#' buffer to the assessment set, ensuring that your analysis data is spatially
+#' separated from the assessment set.
+#' In basic V-fold cross-validation (i.e. no repeats), the number of resamples
+#' is equal to V.
+#'
+#' @details
+#' When `radius` and `buffer` are both `NULL`, `spatial_buffer_vfold_cv`
+#' is equivalent to [rsample::vfold_cv()] and `leave_location_out_cv` is
+#' equivalent to [rsample::group_vfold_cv()].
+#'
+#' @inheritParams rsample::vfold_cv
+#' @inheritParams rsample::group_vfold_cv
+#' @param group A variable in data (single character or name) used to create
+#' folds. For leave location out CV, this should be a variable containing
+#' the locations to group observations by, for leave time out CV the
+#' time blocks to group by, and for leave location and time out the
+#' spatiotemporal blocks to group by.
+#' @inheritParams buffer_indices
+#'
+#' @references
+#'
+#' K. Le Rest, D. Pinaud, P. Monestiez, J. Chadoeuf, and C. Bretagnolle. 2014.
+#' "Spatial leave-one-out cross-validation for variable selection in the
+#' presence of spatial autocorrelation," Global Ecology and Biogeography 23,
+#' pp. 811-820, doi: 10.1111/geb.12161.
+#'
+#' H. Meyer, C. Reudenbach, T. Hengl, M. Katurji, and T. Nauss. 2018.
+#' "Improving performance of spatio-temporal machine learning models using
+#' forward feature selection and target-oriented validation,"
+#' Environmental Modelling & Software 101, pp. 1-9,
+#' doi: 10.1016/j.envsoft.2017.12.001.
+#'
+#' @rdname spatial_vfold
+#'
+#' @examplesIf sf::sf_use_s2() && rlang::is_installed("modeldata")
+#' boston_vfold <- spatial_buffer_vfold_cv(
+#'   boston_canopy,
+#'   buffer = 500,
+#'   radius = NULL
+#' )
+#' data(ames, package = "modeldata")
+#' ames_sf <- sf::st_as_sf(ames, coords = c("Longitude", "Latitude"), crs = 4326)
+#' ames_neighborhoods <- spatial_leave_location_out_cv(ames_sf, Neighborhood)
+#'
+#' @export
+spatial_buffer_vfold_cv <- function(data,
+                                    v = 10,
+                                    radius,
```

Comment:
Should these come before `v`? On the one hand, `v` has a default and these don't, on the other, these arguments come at the end of every other function

Reply:
Yes, since we require users to supply `radius` and `buffer`, let's put them before the arguments with defaults.